### ISSUE TITLE
DietPi-Software | Fails to load global password on Stretch => Buster dist-upgraded systems

### ISSUE THREAD

**Issue by MattL0** [MichaIng/DietPi#2937]
#### Required Information
- DietPi version | 

#!/bin/bash
G_DIETPI_VERSION_CORE=6
G_DIETPI_VERSION_SUB=24
G_DIETPI_VERSION_RC=1
G_GITBRANCH='master'
G_GITOWNER='MichaIng'

- Distro version |

10.0

- Kernel version | 

Linux DietPi-Salon 4.19.50-v7+ #896 SMP Thu Jun 20 16:11:44 BST 2019 armv7l GNU/Linux

- SBC device |  

RPI3 and RPI3b+

#### Steps to reproduce

1. upgrade buster from a stretch install
2. load dietpi-software

#### Expected behaviour

- Should be able to retrieve password?

#### Actual behaviour

- see picture , dietpi fallback passw seems to work.
![Capture](https://user-images.githubusercontent.com/27497561/60060608-72c5db00-96bf-11e9-9976-af1d60df34c5.PNG)

**Comment by MichaIng** [comment]
@MattL0 
Ah you upgraded from Stretch, then this makes sense.

Okay since I don't want to print the encryption keys and stuff here (although it's open source, if one wants to find it 😉), the easiest is:
```
rm /var/lib/dietpi/dietpi-software/.GLOBAL_PW.bin
G_CONFIG_INJECT 'AUTO_SETUP_GLOBAL_PASSWORD=' 'AUTO_SETUP_GLOBAL_PASSWORD=yourPassword' /DietPi/dietpi.txt
```
- Choose yourPassword freely of course.
- Next time DietPi-Software will re-encrypt this password silently.

The issue was that on Buster a new key derivation method is available and if not used, ugly warnings are printed. So some DietPi versions ago we moved Buster systems passwords to the new encryption method. But if you upgrade a system from Stretch to Buster, this is not done, hence DietPi-Software tries to use a key derivation method that does not fit the one of the encrypted password => failure.

I will add the ondemand encryption method upgrade to every DietPi-Software and more current DietPi-Update runs. This is actually not a big deal and allows Buster distro-upgrades then.

**Comment by MattL0** [comment]
thanks :)

**Comment by MattL0** [comment]
Works perfect *

**Comment by MichaIng** [comment]
@MattL0 
Thanks for feedback. I will reopen the issue until I implemented the automatic encryption transition into code.

**Comment by PeterLacknase** [comment]
@MattL0 

Out of curiosity:

> Steps to reproduce
> 1. upgrade buster from a stretch install
> 2. load dietpi-software

What where the exact steps (i.e. commands) you used? Did you change anything else apart from sources.list?

**Comment by MattL0** [comment]
No other step  aappart from changing /etc/apt/sources.list.d/raspi.list with buster instead of stretch.

also did 

apt update 
apt full-upgrade

**Comment by MichaIng** [comment]
Finished: DietPi-Software will now automatically attempt do decrypt the password with the Stretch method, if it fails on Buster. If this succeeds, it will automatically re-encrypt the password with the Buster method.
Changelog: https://github.com/MichaIng/DietPi/commit/15038cf2b710667a4529f230ddc9699b8f213855

Will be merged into Beta 6.25.1
